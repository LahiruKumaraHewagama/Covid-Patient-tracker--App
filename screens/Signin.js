import React, { useState } from "react";
import {
  StyleSheet,
  Text,
  View,
  SafeAreaView,
  TouchableOpacity,
  TextInput,
} from "react-native";
import * as Animatable from "react-native-animatable";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import Feather from "react-native-vector-icons/Feather";
import AsyncStorage from "@react-native-async-storage/async-storage";
// import { AsyncStorage } from "react-native";

function Signin({ navigation }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [loading, setLoading] = useState(false);
  const [errortext, setErrortext] = useState("");

  const AppButton = ({ onPress, title }) => (
    <TouchableOpacity onPress={onPress} style={styles.button}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  );

  const [data, setData] = React.useState({
    secureTextEntry: true,
    isValidUser: true,
  });

  const updateSecureTextEntry = () => {
    setData({
      ...data,
      secureTextEntry: !data.secureTextEntry,
    });
  };

  const signIn = async () => {
    const URL = "http://192.168.8.102:8001/auth/login";

    try {
      const res = await fetch(URL, {
        method: "POST",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          email: email,
          password: password,
        }),
      });
      const response = await res.json();

      // console.log(response.results);

      if (res.status != 200) {
        throw new Error();
      } else {
        if (response.results.token) {
          try {
            await AsyncStorage.setItem("userInfo", JSON.stringify(response));
          } catch (error) {}

          if (response.results.accType === "DOC") {
            navigation.navigate("DoctorDashboard");
          } else {
            navigation.navigate("HospitalAdminDashboard");
          }
        }
      }
    } catch (error) {
      alert("Invalid User!", "email or password is incorrect.", [
        { text: "Okay" },
      ]);
    }
  };

  const handleSubmitPress = () => {
    setErrortext("");
    if (!email) {
      alert("Email can't be empty !");
      return;
    }
    if (!password) {
      alert("Password can't be empty !");
      return;
    }
    signIn();
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.textHeader}>Welcome!</Text>
      </View>
      <Animatable.View style={styles.footer} animation="fadeInUpBig">
        <Text style={styles.textFooter}>Email</Text>
        <View style={styles.action}>
          <FontAwesome name="envelope-square" size={20} />
          <TextInput
            placeholder="Your Email"
            placeholderTextColor="#666666"
            style={styles.textInput}
            autoCapitalize="none"
            onChangeText={(email) => setEmail(email)}
          />
        </View>

        <Text style={[styles.textFooter, { marginTop: 35 }]}>Password</Text>
        <View style={styles.action}>
          <FontAwesome name="lock" size={20} />
          <TextInput
            placeholder="Your Password"
            placeholderTextColor="#666666"
            secureTextEntry={data.secureTextEntry ? true : false}
            style={styles.textInput}
            autoCapitalize="none"
            onChangeText={(password) => setPassword(password)}
          />

          <TouchableOpacity onPress={updateSecureTextEntry}>
            {data.secureTextEntry ? (
              <Feather name="eye-off" color="grey" size={20} />
            ) : (
              <Feather name="eye" color="grey" size={20} />
            )}
          </TouchableOpacity>
        </View>

        <TouchableOpacity>
          <Text style={{ color: "#009387", marginTop: 30 }}>
            Forgot password?
          </Text>
        </TouchableOpacity>

        <AppButton onPress={handleSubmitPress} title={"Sign In"} />
      </Animatable.View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#009387",
  },
  header: {
    flex: 1,
    justifyContent: "flex-end",
    paddingHorizontal: 20,
    paddingBottom: 25,
  },
  footer: {
    flex: 3,
    backgroundColor: "#fff",
    borderTopLeftRadius: 30,
    borderTopRightRadius: 30,
    paddingHorizontal: 20,
    paddingVertical: 50,
  },
  textHeader: {
    color: "#fff",
    fontWeight: "bold",
    fontSize: 30,
  },
  textFooter: {
    color: "#000",
    fontSize: 20,
  },
  action: {
    flexDirection: "row",
    marginTop: 20,
    borderBottomWidth: 1,
    borderBottomColor: "#f2f2f2",
    paddingBottom: 5,
    alignItems: "center",
  },
  actionError: {
    flexDirection: "row",
    marginTop: 10,
    borderBottomWidth: 1,
    borderBottomColor: "#FF0000",
    paddingBottom: 5,
  },
  textInput: {
    flex: 1,
    paddingLeft: 20,
    color: "#000",
  },
  errorMsg: {
    color: "#FF0000",
    fontSize: 14,
  },
  button: {
    marginTop: 100,
    paddingVertical: 10,
    paddingHorizontal: 50,
    backgroundColor: "#009387",
    borderRadius: 50,
    alignItems: "center",
  },
  buttonText: {
    color: "#fff",
    fontSize: 20,
  },
});

export default Signin;
