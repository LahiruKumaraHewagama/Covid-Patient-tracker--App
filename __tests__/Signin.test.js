import React from "react";
import Signin from "../screens/Signin";
import { act, create } from "react-test-renderer";
import { render, fireEvent } from '@testing-library/react-native';

jest.useFakeTimers()
const navigation = { navigate: jest.fn() }


const tree = create(<Signin navigation={navigation} />);

test('snapshot signin page ', async () => {
    expect(tree).toMatchSnapshot();
});

test('Email empty input error massage', async () => {
    global.alert = jest.fn();
    const signInBtn = tree.root.findByProps({ testID: 'signin' }).props;
    act(() => signInBtn.onPress());

    expect(global.alert).toHaveBeenCalledWith("Email can't be empty !");

});

test('Password empty input error massage', async () => {
    global.alert = jest.fn();
    const email = tree.root.findByProps({ testID: 'email_test' }).props;
    act(() => email.onChangeText("lahiru@gmail.com"));

    const signInBtn = tree.root.findByProps({ testID: 'signin' }).props;
    act(() => signInBtn.onPress());

    expect(global.alert).toHaveBeenCalledWith("Password can't be empty !");

});


test('Invalid user error massage', async () => {
    global.alert = jest.fn();
    const email = tree.root.findByProps({ testID: 'email_test' }).props;
    act(() => email.onChangeText("lahiru@gmail.com"));

    const password = tree.root.findByProps({ testID: 'password_test' }).props;
    act(() => password.onChangeText("ABC*"));

    const signInBtn = tree.root.findByProps({ testID: 'signin' }).props;
    act(() => signInBtn.onPress());

    expect(global.alert).toHaveBeenCalledWith("Invalid User!", "email or password is incorrect.", [{"text": "Okay"}])

});




