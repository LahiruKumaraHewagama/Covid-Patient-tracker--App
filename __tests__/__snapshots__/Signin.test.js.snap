// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot signin page  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#009387",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "paddingBottom": 25,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      SIGN IN 
    </Text>
  </View>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 3,
        "opacity": 0,
        "paddingHorizontal": 20,
        "paddingVertical": 50,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#007c7a",
          "fontSize": 16,
        }
      }
    >
      Email Address
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#007c7a",
          "borderRadius": 20,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginTop": 10,
          "padding": 5,
        }
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Your Email"
        placeholderTextColor="#666666"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "paddingLeft": 20,
          }
        }
        testID="email_test"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#007c7a",
            "fontSize": 16,
          },
          Object {
            "marginTop": 35,
          },
        ]
      }
    >
      Password
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#007c7a",
          "borderRadius": 20,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginTop": 10,
          "padding": 5,
        }
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Your Password"
        placeholderTextColor="#666666"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "paddingLeft": 20,
          }
        }
        testID="password_test"
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "marginTop": 30,
          }
        }
      >
        Forgot Your Password?
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#009387",
          "borderColor": "#20d1ce",
          "borderRadius": 10,
          "borderWidth": 2,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 100,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
      testID="signin"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
          }
        }
      >
        Sign In
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
